enemy/enemy_alt.gd

```diff
@@ -0,0 +1,182 @@
+class_name EnemyAlt
+extends CharacterBody2D
+
+signal hit(damage: float)
+signal died
+
+@export var hitbox: CollisionShape2D
+@export var sensor: Area2D
+@export var anim: AnimatedSprite2D
+@export var hit_particle: Resource
+@export var hit_particle_color: Color
+@export var attack_sfx: AudioStreamPlayer2D
+@export var attack_frame_index: int = 2
+
+@export var health: int = 2
+@export var damage: int = 1
+@export var speed: float = 100.0
+@export var coin_reward: int = 1
+
+var _attack_position = Vector2.INF
+var _state = 'idle'
+
+signal entered_attack_hit_frame
+
+
+func is_alive() -> bool:
+	return health > 0
+
+
+func set_difficulty(difficulty: float) -> void:
+	health += ceil(difficulty)
+	damage += ceil(difficulty * 0.5)
+	speed += ceil(difficulty * 0.5)
+	coin_reward += floor(difficulty * 1.5)
```

Comment:
this is not currently used but could be called in the `enemy_spawn` script after instantiating a new enemy.